l1-contracts/contracts/bridge/L1NativeTokenVault.sol

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IL1NativeTokenVault} from "./interfaces/IL1NativeTokenVault.sol";
+import {ReentrancyGuard} from "../common/ReentrancyGuard.sol";
+import {IL1StandardAsset} from "./interfaces/IL1StandardAsset.sol";
+
+import {IL1SharedBridge} from "./interfaces/IL1SharedBridge.sol";
+import {ETH_TOKEN_ADDRESS, NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS} from "../common/Config.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @dev Vault holding L1 native ETH and ERC20 tokens bridged into the hyperchains.
+/// @dev Designed for use with a proxy for upgradability.
+contract L1NativeTokenVault is
```

Comment:
BTW, why do we have a different interface for L1 native token vault? I recall from the previous discussions the L1 and L2 native vaults should be basically the same or am I missing something

Reply:
Good point.
Hereby, we simply support the same logic that was before.
But we could refactor to aim for native l2 tokens being sent to l1 for example

Reply:
For now we figured sending L2->L1 tokens would be out of scope to keep it small.